frontends/web/src/new_front/components/CreateSamples/CreateSamples/AnnotationInterfaces/Contexts/SelectMultipleTextMultipleTags.tsx

```diff
@@ -0,0 +1,312 @@
+    generative_context?.artifacts?.tags?.forEach((tag: any) => {
+      let color = generateLightRandomColor();
+      while (colors.includes(color)) {
+        color = generateLightRandomColor();
+      }
+      colors.push(color);
+      tempTags.push({
+        value: tag?.display_label,
+        color: color,
+        back_label: tag?.back_label,
+      });
+      tempTagColors[tag.back_label] = color;
+    });
+    setLocalTags(tempTags);
+    setTagColors(tempTagColors);
+    if (field_names_for_the_model?.preselected_tag) {
+      setPreferedTag(field_names_for_the_model.preselected_tag);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (preferedTag) {
+      setTagSelection(localTags.find((tag: any) => tag.value === preferedTag));
+      handleSubmit(
+        localTags.find((tag: any) => tag.value === preferedTag)?.back_label,
+      );
+    }
+  }, [preferedTag]);
+
+  const handleSubmit = async (value: string | null) => {
+    !value && (value = tagSelection.back_label);
+    submitButton && (submitButton.hidden = false);
+    const bringContext = await post(
+      "/context/get_random_context_from_key_value/",
+      {
+        key_name: field_names_for_the_model?.tag_name_search,
+        key_value: value,
+      },
+    );
+    if (response.ok) {
+      !bringContext &&
+        Swal.fire({
+          title: instruction?.context_alert_title,
+          text: instruction?.context_alert_text,
+          icon: "warning",
+          confirmButtonText: "Ok",
+        }).then(() => {
+          handleSubmit(field_names_for_the_model?.default_tag);
+        });
+      console.log("bringContext", bringContext);
+      setText(bringContext?.content);
+      setContextId(bringContext?.id);
+      setRealRoundId(bringContext?.round_id);
+    }
+  };
+
+  const handleSelectAll = async () => {
+    const tokens = text?.split(" ");
+    setSelectionInfo([
+      {
+        start: 0,
+        end: tokens?.length,
+        tag: tagSelection.back_label,
+        tokens: tokens,
+        color: tagColors[tagSelection.back_label],
+      },
+    ]);
+  };
+
+  const handleSubmitExample = async () => {
+    const newSelectionInfo = cleanUpSelection(selectionInfo, "color");
+    const response = await post("/example/create_example/", {
+      context_id: contextId,
+      user_id: userId,
+      input_json: { labels: newSelectionInfo },
+      text: text,
+      task_id: taskId,
+      round_id: realRoundId,
+      increment_context: true,
+    });
+    if (response.ok) {
+      Swal.fire({
+        title: "Success",
+        text: "The data has been saved",
+        icon: "success",
+        confirmButtonText: "Ok",
+      }).then(() => {
+        handleSubmit(null);
+      });
+    }
+  };
+
+  const handleChange = (value: any) => {
+    setSelectionInfo(value);
+  };
+
+  return (
+    <AnnotationInstruction
+      placement="top"
+      tooltip={
+        instruction?.preselection_tag ||
+        "Select the tag and the text according to the tag"
```

Comment:
We should try to include tooltip instructions in the yaml so they can be easily modified directly by the task owner.

Reply:
That instruction comes from the Yaml file, the select the tag is in case that instruction isn't specified in it